src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/ServiceBasedViewComponentActivator.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.ViewComponents
+{
+    public class ServiceBasedViewComponentActivator : IViewComponentActivator
+    {
+        public object Create(ViewComponentContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var controllerType = context.ViewComponentDescriptor.TypeInfo.AsType();
+
+            return context.ViewContext.HttpContext.RequestServices.GetRequiredService(controllerType);
+        }
+
+        public virtual void Release(ViewComponentContext context, object viewComponent)
+        {
+        }
```

Comment:
Our service provider doesn't do anything on this method, so I'm not sure we want to add that. I'm sticking to the /// <inheritdoc /> comment here.